plugin.video.carnet-meduza/resources/lib/carnet-meduza.py

```diff
@@ -0,0 +1,264 @@
+def category_videos(name,current_page_number):
+	"""Returns list of vidoes (25 per page) inside selected category"""
+	for category in categories:
+		if category[lang_api_prop] == name:
+			skip = int(current_page_number) * 25
+			category_id = category['ID']
+			request_category_videos = requests.get(api_base_url + 'category/?id=' + category_id + '&skip=' + str(skip) + '&uid=' + api_key)
+			videos = request_category_videos.json()
+        return videos 
+
+def search_videos():
+	"""Returns search results vidoes"""
+        search_heading = addon.getLocalizedString(30206)
+	keyboard = xbmc.Keyboard()
+        keyboard.setHeading(search_heading)
+	keyboard.doModal()
+	if (keyboard.isConfirmed()):
+		query = keyboard.getText()
+		if query is None or len(str(query)) == 0:
+			return
+	else:
+		return
+	sort_videos = addon.getSetting('sort_videos')
+	asc_order = addon.getSetting('asc_order')	
+	request_search_videos = requests.get(api_base_url + 'videos/?query=' + query + '&order=' + str(sort_videos) + '&uid=' + api_key + '&asc=' + str(int(bool(asc_order))))
+	videos = request_search_videos.json()
+	return videos 
+
+def category_video_count(category_id):
+	"""Returns total number of videos inside the category"""
+	request_category_video_count = requests.get(api_base_url + 'category/count/?id=' + category_id + '&uid=' + api_key)
+	category_video_count = request_category_video_count.json()
+	return category_video_count
+
+def video_url_description(video_id):
+	"""Returns video description"""
+	request_video_url_description = requests.get(api_base_url + 'video/?id=' + video_id + '&uid=' + api_key).json()
+	extract_keys = ['stream_url', 'opis']
+	video_url_description = dict((k, request_video_url_description[k]) for k in extract_keys if k in request_video_url_description)
+	return video_url_description
+
+def recommended_videos():
+	"""Returns list of recommended videos. The number of videos can be defined inside settings. Default is 20."""
+	num_recommends = xbmcaddon.Addon('plugin.video.carnet-meduza').getSetting('num_recommends')
+	request_recommended_videos = requests.get(api_base_url + 'recommended/?number=' + str(num_recommends) + '&uid=' + api_key)
+	videos = request_recommended_videos.json()
+	return videos 
+
+def list_search_or_recommended_videos(videos):
+	if videos == None:
+		return
+	else:	
+		for video in videos:
+			category_id = video['ID_kategorija']
+			name = video['naslov']
+			video_id = video['ID']
+			genre = video['kategorija']
+			duration = reduce(lambda x, y: x*60+y, [int(i) for i in (video['trajanje'].replace(':',',')).split(',')])
+			video_info = video_url_description(video_id)
+			try:
+				url = video_info['stream_url'].encode('utf-8') + '|User-Agent=Mozilla/5.0&Referer=https://meduza.carnet.hr'
+				description = video_info['opis'].encode('utf-8')
+			except KeyError:
+				url = ''
+				description = addon.getLocalizedString(30207)
+				pass		
+			image = video['slika']
+			li = xbmcgui.ListItem(name, iconImage=image)
```

Comment:
The use of iconImage initializer is deprecated. Please use setArt. See http://mirrors.kodi.tv/docs/python-docs/16.x-jarvis/xbmcgui.html#ListItem for more info.